src/org/traccar/database/DriversManager.java

```diff
@@ -17,147 +17,53 @@
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.LinkedList;
 import java.util.Map;
-import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 
-import org.traccar.Context;
 import org.traccar.helper.Log;
-import org.traccar.model.Device;
 import org.traccar.model.DeviceDriver;
 import org.traccar.model.Driver;
 import org.traccar.model.DriverPermission;
 import org.traccar.model.GroupDriver;
+import org.traccar.model.Identifiable;
 
-public class DriversManager {
+public class DriversManager extends ExtendedObjectManager {
 
-    private final DataManager dataManager;
-
-    private final Map<Long, Driver> drivers = new ConcurrentHashMap<>();
     private final Map<String, Driver> driversByUniqueId = new ConcurrentHashMap<>();
-    private final Map<Long, Set<Long>> deviceDrivers = new ConcurrentHashMap<>();
-    private final Map<Long, Set<Long>> deviceDriversWithGroups = new ConcurrentHashMap<>();
-    private final Map<Long, Set<Long>> groupDrivers = new ConcurrentHashMap<>();
-    private final Map<Long, Set<Long>> userDrivers = new ConcurrentHashMap<>();
 
     public DriversManager(DataManager dataManager) {
-        this.dataManager = dataManager;
-        refreshDrivers();
-    }
-
-    public Set<Long> getUserDrivers(long userId) {
-        if (!userDrivers.containsKey(userId)) {
-            userDrivers.put(userId, new HashSet<Long>());
-        }
-        return userDrivers.get(userId);
-    }
-
-    public Set<Long> getGroupDrivers(long groupId) {
-        if (!groupDrivers.containsKey(groupId)) {
-            groupDrivers.put(groupId, new HashSet<Long>());
-        }
-        return groupDrivers.get(groupId);
-    }
-
-    public Set<Long> getDeviceDrivers(long deviceId) {
-        return getDeviceDrivers(deviceDrivers, deviceId);
-    }
-
-    public Set<Long> getAllDeviceDrivers(long deviceId) {
-        return getDeviceDrivers(deviceDriversWithGroups, deviceId);
-    }
-
-    private Set<Long> getDeviceDrivers(Map<Long, Set<Long>> deviceDrivers, long deviceId) {
-        if (!deviceDrivers.containsKey(deviceId)) {
-            deviceDrivers.put(deviceId, new HashSet<Long>());
-        }
-        return deviceDrivers.get(deviceId);
+        super(dataManager, Driver.class, DriverPermission.class, DeviceDriver.class, GroupDriver.class);
+        refreshItems();
+        refreshExtendedPermissions();
     }
 
-    public final void refreshDrivers() {
-        if (dataManager != null) {
+    @Override
+    public void refreshItems() {
+        if (getDataManager() != null) {
             try {
-                drivers.clear();
-                driversByUniqueId.clear();
-                for (Driver driver : dataManager.getDrivers()) {
-                    drivers.put(driver.getId(), driver);
-                    driversByUniqueId.put(driver.getUniqueId(), driver);
+                clearItems();
+                for (Identifiable item : getDataManager().getObjects(getBaseClass())) {
```

Comment:
The meaning of words are perfectly reflects the idea. One has ID, second has attributes and can be easily extended.
But agree nouns fits more for class naming.